core/services/workflows/syncer/fetcher.go

```diff
@@ -2,41 +2,111 @@ package syncer
 
 import (
 	"context"
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"net/http"
 	"strings"
 
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
 	"github.com/smartcontractkit/chainlink/v2/core/capabilities/webapi"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
 	ghcapabilities "github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/common"
 )
 
-func NewFetcherFunc(
-	lggr logger.Logger,
-	och *webapi.OutgoingConnectorHandler) FetcherFunc {
-	return func(ctx context.Context, url string) ([]byte, error) {
-		payloadBytes, err := json.Marshal(ghcapabilities.Request{
-			URL:    url,
-			Method: http.MethodGet,
-		})
-		if err != nil {
-			return nil, fmt.Errorf("failed to marshal fetch request: %w", err)
-		}
+const (
+	defaultFetchTimeoutMs = 20_000
+)
 
-		messageID := strings.Join([]string{ghcapabilities.MethodWorkflowSyncer, url}, "/")
-		resp, err := och.HandleSingleNodeRequest(ctx, messageID, payloadBytes)
-		if err != nil {
-			return nil, err
+type FetcherService struct {
+	services.StateMachine
+	lggr    logger.Logger
+	och     *webapi.OutgoingConnectorHandler
+	wrapper gatewayConnector
+}
+
+type gatewayConnector interface {
+	GetGatewayConnector() connector.GatewayConnector
+}
+
+func NewFetcherService(lggr logger.Logger, wrapper gatewayConnector) *FetcherService {
+	return &FetcherService{
+		lggr:    lggr.Named("FetcherService"),
+		wrapper: wrapper,
+	}
+}
+
+func (s *FetcherService) Start(ctx context.Context) error {
+	return s.StartOnce("FetcherService", func() error {
+		connector := s.wrapper.GetGatewayConnector()
+
+		outgoingConnectorLggr := s.lggr.Named("WorkflowSyncer")
+
+		webAPIConfig := webapi.ServiceConfig{
+			RateLimiter: common.RateLimiterConfig{
+				GlobalRPS:      100.0,
+				GlobalBurst:    100,
+				PerSenderRPS:   100.0,
+				PerSenderBurst: 100,
+			},
 		}
 
-		lggr.Debugw("received gateway response", "resp", resp)
-		var payload ghcapabilities.Response
-		err = json.Unmarshal(resp.Body.Payload, &payload)
+		och, err := webapi.NewOutgoingConnectorHandler(connector,
+			webAPIConfig,
+			capabilities.MethodWorkflowSyncer, outgoingConnectorLggr)
 		if err != nil {
-			return nil, err
+			return fmt.Errorf("could not create outgoing connector handler: %w", err)
 		}
 
-		return payload.Body, nil
+		s.och = och
+		return och.Start(ctx)
+	})
+}
+
+func (s *FetcherService) Close() error {
+	return nil
```

Comment:
Why not call `s.StopOnce` for completeness?

Also, does `s.och` need to be closed?